projects/RabbitMQ.Client/client/impl/ModelBase.cs

```diff
@@ -910,7 +916,44 @@ public void BasicPublish<TProperties>(string exchange, string routingKey, in TPr
             }
 
             var cmd = new BasicPublish(exchange, routingKey, mandatory, default);
-            ModelSend(in cmd, in basicProperties, body);
+            using (Activity sendActivity = RabbitMQActivitySource.Send(cmd, body.Length))
+            {
+                if (sendActivity != null)
+                {
+                    if (sendActivity.IsAllDataRequested)
+                    {
+                        if (!string.IsNullOrEmpty(basicProperties.CorrelationId))
+                        {
+                            sendActivity.AddTag("messaging.conversation_id", basicProperties.CorrelationId);
+                        }
+                    }
+
+                    // This activity is marked as recorded, so let's propagate the trace and span ids.
+                    if (sendActivity.ActivityTraceFlags == ActivityTraceFlags.Recorded)
+                    {
+                        BasicProperties props = default;
+                        if(basicProperties is BasicProperties properties)
+                        {
+                            props = properties;
+                        }
+                        else if(basicProperties is EmptyBasicProperty)
+                        {
+                            props = new BasicProperties();
+                        }
+                        else if(basicProperties is ReadOnlyBasicProperties readOnlyProps)
+                        {
+                            props = new BasicProperties(readOnlyProps);
+                        }
+
+                        props.Headers ??= new Dictionary<string, object>();
+                        props.Headers["otel.activity_id"] = sendActivity.Id;
```

Comment:
I'd strongly recommend using the W3CTraceContext spec by default.
This is ideally done by using the Propagators from the OpenTelemetry.API.

Reply:
Will take a look at that :)